webrtc.html

```diff
@@ -11123,6 +11127,161 @@ <h4>MediaTrackSupportedConstraints, MediaTrackCapabilities,
       </section>
     </section>
   </section>
+  <section>
+    <h2>Direct Connection Permission API</h2>
+    <p>The Direct Connection Permission API allows a web application to <a>request permission to use</a> a less strict policy regarding sharing local addresses as defined in [[RTCWEB-IP-HANDLING]]. This API and the behavioral requirements of the user agent ensure that permission can be requested in a use case neutral way. Once permission has been granted, it provides explicit user consent as required by [[RTCWEB-IP-HANDLING]].</p>
+    <p>The <dfn data-lt="best-available-mode">best available IP handling mode</dfn> the user agent can offer SHOULD be Mode 1 (enumerate all addresses) and the <dfn data-lt="default-mode">default IP handling mode</dfn> the user agent uses by default SHOULD be Mode 2 (default route and associated local addresses). However, the user agent MAY choose different modes determined by the user agents preferences.</p>
+    <section>
+      <h3>RTCPeerConnection Interface Extensions</h3>
+      <p>The Direct Connection Permission API extends the <code><a>RTCPeerConnection</a></code> interface as described below.</p>
+      <div>
+        <pre class="idl">partial interface RTCPeerConnection {
+    static void registerDirectConnectionInterest ();
```

Comment:
The rationale is that:

1. A peer connection is invisible in terms of UI, so there's no way to distinguish them for the user.
2. Granting a permission would have an effect on the same origin, so all will be affected.

Reply:
That's defining the API based on currently imaginable UI.
I don't think that's a wise design principle.
One could imagine other ways to determine permission, for instance that signed apps have different permissions from unsigned apps, or even that Javscript running inside some future sandbox (other than an iframe) had yet another means of deciding on permission.

We should shape the API so that an app developer can give information to the UA that is relevant to what he's working on; the developer of a conferencing system should not have to worry about what the ad-supported clicktracker on the same page is telling the UA (and vice versa).

Reply:
Okay, I think that is reasonable. Any further thoughts, @jan-ivar?

Reply:
If we go with *normal method*, I'd like to add a note that the UA may consider to treat it as a request towards that particular `RTCPeerConnection` instance or increase the scope to the same origin.